appstudio-controller/controllers/appstudio.redhat.com/devsandbox_deployment_controller.go

```diff
@@ -0,0 +1,213 @@
+			"SpaceRequest.Namespace", spacerequest.Namespace)
+	}
+
+	log.Info("A DeploymentTarget for the SpaceRequest exists", "DeploymentTarget.Name", dt.Name, "Namespace", dt.Namespace)
+
+	dt.Status.Phase = applicationv1alpha1.DeploymentTargetPhase_Available // set phrase to "Available"
+
+	return ctrl.Result{}, nil
+}
+
+// findMatchingDTForSpaceRequest tries to find a DT that matches the given SpaceRequest in a namespace.
+func findMatchingDTForSpaceRequest(ctx context.Context, k8sClient client.Client, spacerequest *codereadytoolchainv1alpha1.SpaceRequest) (*applicationv1alpha1.DeploymentTarget, error) {
+	if dtc, _ := findMatchingDTCForSpaceRequest(ctx, k8sClient, spacerequest); dtc != nil {
+		dtList := applicationv1alpha1.DeploymentTargetList{}
+		opts := []client.ListOption{
+			client.InNamespace(spacerequest.Status.NamespaceAccess[0].Name),
+		}
+
+		if err := k8sClient.List(ctx, &dtList, opts...); err != nil {
+			return nil, err
+		}
+
+		var dt *applicationv1alpha1.DeploymentTarget
+		for i, d := range dtList.Items {
+			if dtc.Name == string(d.Spec.ClaimRef) {
+				dt = &dtList.Items[i]
+				return dt, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func isLabelFound(spacerequest *codereadytoolchainv1alpha1.SpaceRequest) bool {
+	if spacerequest.Labels == nil || len(spacerequest.Labels) == 0 {
+		return false
+	}
+
+	_, found := spacerequest.Labels[deploymentTargetClaimLabel]
+	return found
+}
+
+// findMatchingDTCForSpaceRequest tries to find the DTC that matches a given SpaceRequest.
+func findMatchingDTCForSpaceRequest(ctx context.Context, k8sClient client.Client, spacerequest *codereadytoolchainv1alpha1.SpaceRequest) (*applicationv1alpha1.DeploymentTargetClaim, error) {
+	if !isLabelFound(spacerequest) {
+		return nil, fmt.Errorf("no 'appstudio.openshift.io/dtc' label is set for spacerequest %s %s", spacerequest.Name, spacerequest.Namespace)
+	}
+	dtcList := applicationv1alpha1.DeploymentTargetClaimList{}
+	opts := []client.ListOption{
+		client.InNamespace(spacerequest.Status.NamespaceAccess[0].Name),
+	}
+
+	if err := k8sClient.List(ctx, &dtcList, opts...); err != nil {
+		return nil, err
+	}
+
+	if len(dtcList.Items) > 0 {
+		for _, d := range dtcList.Items {
+			if d.Name == spacerequest.Labels[deploymentTargetClaimLabel] {
+				return &d, nil
+			}
+		}
+	}
+
+	return nil, fmt.Errorf("no dtc is found for spacerequest %s %s", spacerequest.Name, spacerequest.Namespace)
+}
+
+// NewDeploymentTarget creates a new DeploymentTarget using the provided info.
+func NewDeploymentTarget(deploymentTargetClassName applicationv1alpha1.DeploymentTargetClassName, namespace string, clusterAPIURL string, secretRef string, dtcName string) *applicationv1alpha1.DeploymentTarget {
+	dtName := dtcName + "-dt"
+	deploymentTarget := &applicationv1alpha1.DeploymentTarget{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: dtName + "-",
+			Namespace:    namespace,
```

Comment:
I think that the DeploymentTarget's `namespace` should be different from the `DefaultNamespace`, because the DT resource should be in the user's namespace, while the `spacerequest.Status.NamespaceAccess[0].Name` references the temporary namespace that was provisioned.

Reply:
Thanks, looks good!